server/backends/blobstore/blobstore.go

```diff
@@ -537,15 +543,15 @@ func (a *AwsS3BlobStore) WriteBlob(ctx context.Context, blobName string, data []
 		return 0, err
 	}
 	start := time.Now()
-	n, err := a.upload(ctx, blobName, compressedData)
+	n, err := a.upload(ctx, blobPath(blobName), compressedData)
```

Comment:
`upload()` is already calling `blobPath()`, so this line should be reverted to avoid double-prefixing:

```suggestion
	n, err := a.upload(ctx, blobName, compressedData)
```